src/GenSymIO.chpl

```diff
@@ -1928,146 +1949,101 @@ module GenSymIO {
+             * For the special case of this being the first locale, set the shuffleLeftIndex 
              * to -1 since there is no previous locale that has an incomplete string at the
              * end that will require chars sliced from locale 0 to complete. If there is one
              * null uint(8) char that is not at the end of the values array, this is the 
-             * trailingSliceIndex for the first locale.
+             * shuffleRightIndex for the first locale.
              */
             if idx == 0 {
-                if leadingSliceIndices[idx] > -1 {
-                    trailingSliceIndices[idx] = leadingSliceIndices[idx];
+                if shuffleLeftIndices[idx] > -1 {
+                    shuffleRightIndices[idx] = shuffleLeftIndices[idx];
                 }
-                leadingSliceIndices[idx] = -1;
+                shuffleLeftIndices[idx] = -1;
             }
             
             /*
-             * For the special case of this being the last locale, set the trailingSliceIndex 
-             * to -1 since there is no next locale to shuffle trailing slice to.
+             * For the special case of this being the last locale, set the shuffleRightIndex 
+             * to -1 since there is no next locale to shuffle a trailing slice to.
              */
             if isLastLocale(idx) {
-                trailingSliceIndices[idx] = -1;
+                shuffleRightIndices[idx] = -1;
             }
         }
     }
     
     /*
-     * Processes a local Strings slice into (1) a uint(8) values list for use in methods 
-     * that finalize the values array elements following any shuffle operations and (2)
-     * a segments list representing starting indices for each string in the values list.
+     * Adjusts for the shuffling of a leading char sequence to the previous locale by 
+     * slicing leading chars that compose a string started in the previous locale and 
+     * returning a new char array.
      */
-    private proc sliceToValuesAndSegments(rawChars) throws {
-        var charList: list(uint(8), parSafe=false);
-        var indices: list(int, parSafe=false);
-
-        //initialize segments with index to first char in values
-        indices.append(0);
-        
-        for (value, i) in zip(rawChars, 0..rawChars.size-1) do {
-            charList.append(value:uint(8));
-            /*
-             * If the char is the null uint(8) char, check to see if it is the 
-             * last char. If not, added to the indices. If it is the last char,  
-             * don't add, because it is the correct ending char for a Strings 
-             * values array to be written to a locale.
-             */ 
-            if value == NULL_STRINGS_VALUE && i < rawChars.size-1 {
-                indices.append(i+1);
-            }
-        }
-
-        return (charList, indices);
-    }
+    private proc adjustCharArrayForLeadingSlice(sliceIndex, charArray, last, idx) throws { 
```

Comment:
Note: both here and in `adjustForLeftShuffle`, the `idx` parameter is never used and can probably be removed.

Reply:
@glitch It was used for logging, will remove it

Reply:
Fixed, checked in